### ISSUE TITLE
network.LookupSubnet causes segmentation violation with azure provider

### ISSUE THREAD

**Issue by thomas-b-jackson** [pulumi/pulumi#5841]
I can use terraform to obtain the ID of an azure subnet using a `data` resource, but the comparable pulumi code in go results in a segmentation violation

## Expected Behavior

1. run:
    `terraform apply`
1. correct subnet_id is in output
1. run: 
    `pulumi up`
1. correct subnet id is in output

## Current Behavior

1. run:
    `terraform apply`
1. correct subnet_id is in output
1. run: 
    `pulumi up`
1. segmentation fault

```
pulumi:pulumi:Stack (azure-vm-dev):
    panic: runtime error: invalid memory address or nil pointer dereference
    [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xabb15d]
    goroutine 1 [running]:
    github.com/pulumi/pulumi/sdk/v2/go/pulumi.(*Context).Invoke(0xc00024c200, 0x2c1da41, 0x21, 0x2541a00, 0xc00024f500, 0x2541a40, 0xc0000ad760, 0xc00029fbc0, 0x1, 0x1, ...)
        /home/tomj/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.9.2/go/pulumi/context.go:164 +0x17d
    github.com/pulumi/pulumi-azure/sdk/v3/go/azure/network.LookupSubnet(0xc00024c200, 0xc00024f500, 0xc00029fbc0, 0x1, 0x1, 0xc000259790, 0xc0002e0000, 0x0)
        /home/tomj/go/pkg/mod/github.com/pulumi/pulumi-azure/sdk/v3@v3.32.0/go/azure/network/getSubnet.go:39 +0xaf
    main.main.func1(0xc00024c200, 0x2c111e6, 0x13)
        /home/tomj/go/vm/azure/main.go:11 +0xaa
    github.com/pulumi/pulumi/sdk/v2/go/pulumi.RunWithContext(0xc00024c200, 0x2c433b8, 0xc00003a12f, 0x8)
        /home/tomj/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.9.2/go/pulumi/run.go:101 +0x1ff
    github.com/pulumi/pulumi/sdk/v2/go/pulumi.RunErr(0x2c433b8, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/tomj/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.9.2/go/pulumi/run.go:82 +0x20a
    github.com/pulumi/pulumi/sdk/v2/go/pulumi.Run(0x2c433b8, 0x0, 0x0, 0x0)
        /home/tomj/go/pkg/mod/github.com/pulumi/pulumi/sdk/v2@v2.9.2/go/pulumi/run.go:39 +0x59
    main.main()
        /home/tomj/go/vm/azure/main.go:9 +0x3e
    exit status 2

    error: an unhandled error occurred: program exited with non-zero exit code: 1
```

## Steps to Reproduce

main.go is as follows ...

 ```
package main

import (
	"github.com/pulumi/pulumi-azure/sdk/v3/go/azure/network"
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		subnetNetwork, err := network.LookupSubnet(ctx, &network.LookupSubnetArgs{
			Name:               "clusternodes",
			ResourceGroupName:  "npeccpdevwu2RgGolduck",
			VirtualNetworkName: "npeccpdevwu2RgGolduckVPC",
		}, nil)
		if err != nil {
			return err
		}
		ctx.Export("subnetId", pulumi.String(subnetNetwork.Id))
		return nil
	})
}
```

main.tf is as follows:
```
data "azurerm_subnet" "subnet_network" {
  name                   = "clusternodes"
  resource_group_name    = "npeccpdevwu2RgGolduck"
  virtual_network_name   = "npeccpdevwu2RgGolduckVPC"
}

output "subnet_id" {
  value = data.azurerm_subnet.subnet_network.id
}

```

## Context (Environment)

Blocks me from building in azure using go. All of our vm builds are expected to use the same subnet.

**Comment by stack72** [comment]
Hi @thomas-b-jackson 

Apologies you are facing this issue - I am almost positive the issue is due to you passing nil to the resource function 

Can you try and omit that? We will work to ensure we handle that better

Paul

**Comment by stack72** [comment]
Hi @thomas-b-jackson 

On further investigation - I am going to close this out as this is fixed in pulumi v2.10.0. I was able to upgrade to that specific version and ensure the fix is in place

Paul

**Comment by thomas-b-jackson** [comment]
@stack72 , note that I got the code from https://www.pulumi.com/tf2pulumi/
